### PR TITLE
perf(CI): use the first stage for unit tests on Jenkins

### PR DESCRIPTION
**Pull request by skateman**
Should make things faster by ~40 seconds as we would not need to install any packages, just some gems.

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: scripts/unit_test.sh [hunk 1]
```diff
@@ -1,12 +1,16 @@
 #!/bin/bash
 
+if [[ -z "$ghprbPullId" ]]; then
+  echo "Not running from a context of a PR"
+  exit 1
+fi
+
```
Comment (Victoremepunto) [inline scripts/unit_test.sh:7]: use `is_pr_or_mr_build` from `build_deploy_common`

Comment (skateman) [reply scripts/unit_test.sh:7]: How do I pull in build_deploy_common here? By calling `source`?

Comment (skateman) [reply scripts/unit_test.sh]: Also
```
shellcheck scripts/unit_test.sh                                                                                        

In scripts/unit_test.sh line 11:
IMAGE_TAG="pr-${ghprbPullId}-$(git rev-parse --short=7 HEAD)"
              ^------------^ SC2154 (warning): ghprbPullId is referenced but not assigned.

For more information:
  https://www.shellcheck.net/wiki/SC2154 -- ghprbPullId is referenced but not...

```

File: scripts/unit_test.sh [hunk 2]
```diff
@@ -39,6 +37,14 @@ if ! COMPLIANCE_POD_ID=$(podman pod create --name "$POD_NAME"); then
     exit 1
 fi
 
+# Make sure the build stage image is available
+SERVICE_IMAGE=$(podman images --filter "label=BUILD_STAGE_OF=${IMAGE_TAG}" -q)
```
[the remark this answers is not in the document]
Comment (skateman) [reply scripts/unit_test.sh]: I added a `head -1` to make you happy, it automatically sorts based on creation date, so we always get the latest image.

**Pull request merged**